versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/StringLogicImpl.java

```diff
@@ -68,6 +86,53 @@ public StringObj updateString(StringValue previousValue, String contentType, Str
     return updateString(previousValue, contentType, stringValue.getBytes(StandardCharsets.UTF_8));
   }
 
+  @Override
+  public StringValue updateStringOnRef(
+      Reference reference,
+      StoreKey storeKey,
+      Consumer<Builder> commitEnhancer,
+      String contentType,
+      byte[] stringValueUtf8)
+      throws ObjNotFoundException,
+          CommitConflictException,
+          RefNotFoundException,
+          RefConditionFailedException {
+    CommitLogic commitLogic = commitLogic(persist);
+    IndexesLogic indexesLogic = indexesLogic(persist);
+    CommitObj head = commitLogic.headCommit(reference);
+    StoreIndex<CommitOp> index = indexesLogic.buildCompleteIndexOrEmpty(head);
+    StoreIndexElement<CommitOp> existingElement = index.get(storeKey);
+    ObjId existingValueId = null;
+    UUID existingContentId = null;
+    StringValue existing = null;
+    if (existingElement != null) {
+      CommitOp op = existingElement.content();
+      if (op.action().exists()) {
+        existingValueId = op.value();
+        existingContentId = op.contentId();
+        existing = existingValueId != null ? fetchString(requireNonNull(existingValueId)) : null;
+      }
+    }
+    if (existingContentId == null) {
```

Comment:
This makes updating repo-descriptions of already existing repos impossible (those have no CID for repo-description).

Mabe replacing the `if` with a simple `else` helps here.

Should also have some comment to explain the logic wrt repo-desc + repo-config